java/statestore-lambda/src/main/java/sleeper/statestore/committer/lambda/StateStoreCommitterLambda.java

```diff
@@ -58,24 +61,52 @@ public class StateStoreCommitterLambda implements RequestHandler<SQSEvent, SQSBa
     private final StateStoreCommitRequestDeserialiser serDe = new StateStoreCommitRequestDeserialiser();
 
     public StateStoreCommitterLambda() {
-        this(connectToAws());
-    }
+        AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+        AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.defaultClient();
+        String s3Bucket = System.getenv(CONFIG_BUCKET.toEnvironmentVariable());
+
+        InstanceProperties instanceProperties = new InstanceProperties();
+        instanceProperties.loadFromS3(s3Client, s3Bucket);
+        Configuration hadoopConf = HadoopConfigurationProvider.getConfigurationForLambdas(instanceProperties);
 
-    public StateStoreCommitterLambda(StateStoreCommitter committer) {
-        this.committer = committer;
+        TablePropertiesProvider tablePropertiesProvider = new TablePropertiesProvider(instanceProperties, s3Client, dynamoDBClient);
+        StateStoreProvider stateStoreProvider = new StateStoreProvider(instanceProperties, s3Client, dynamoDBClient, hadoopConf);
+        this.committer = new StateStoreCommitter(
+                CompactionJobStatusStoreFactory.getStatusStore(dynamoDBClient, instanceProperties),
+                IngestJobStatusStoreFactory.getStatusStore(dynamoDBClient, instanceProperties),
+                stateStoreProvider.byTableId(tablePropertiesProvider),
+                key -> s3Client.getObjectAsString(instanceProperties.get(DATA_BUCKET), key),
+                Instant::now);
     }
 
     @Override
     public SQSBatchResponse handleRequest(SQSEvent event, Context context) {
         Instant startTime = Instant.now();
         LOGGER.info("Lambda started at {}", startTime);
         List<BatchItemFailure> batchItemFailures = new ArrayList<>();
-        for (SQSMessage message : event.getRecords()) {
+        List<SQSMessage> messages = event.getRecords();
+        PollWithRetries throttlingRetries = PollWithRetries.builder()
+                .pollIntervalAndTimeout(Duration.ofSeconds(5), Duration.ofMinutes(10))
+                .applyMaxPollsOverall()
+                .build();
+        for (int i = 0; i < messages.size(); i++) {
+            SQSMessage message = messages.get(i);
             LOGGER.info("Found message: {}", message.getBody());
             StateStoreCommitRequest request = serDe.fromJson(message.getBody());
             try {
-                committer.apply(request);
-            } catch (RuntimeException | StateStoreException e) {
+                DynamoDBUtils.retryOnThrottlingException(throttlingRetries, () -> {
+                    try {
+                        committer.apply(request);
+                    } catch (StateStoreException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            } catch (InterruptedException e) {
+                messages.subList(i, messages.size())
```

Comment:
Potentially worth having a logger.error message here so as can determine which of the catch exeception were triggered on any failure

Reply:
Done, thanks